pkg/services/sqlstore/star.go

```diff
@@ -7,17 +7,17 @@ import (
 	"github.com/grafana/grafana/pkg/models"
 )
 
-func init() {
-	bus.AddHandler("sql", StarDashboard)
-	bus.AddHandler("sql", UnstarDashboard)
-	bus.AddHandler("sql", GetUserStars)
-	bus.AddHandlerCtx("sql", IsStarredByUserCtx)
+func (ss *SQLStore) addStarQueryAndCommandHandlers() {
+	bus.AddHandlerCtx("sql", ss.StarDashboard)
+	bus.AddHandlerCtx("sql", ss.UnstarDashboard)
+	bus.AddHandlerCtx("sql", ss.GetUserStars)
+	bus.AddHandlerCtx("sql", ss.IsStarredByUserCtx)
 }
 
-func IsStarredByUserCtx(ctx context.Context, query *models.IsStarredByUserQuery) error {
-	return withDbSession(ctx, x, func(dbSession *DBSession) error {
+func (ss *SQLStore) IsStarredByUserCtx(ctx context.Context, query *models.IsStarredByUserQuery) error {
+	return ss.WithTransactionalDbSession(ctx, func(sess *DBSession) error {
```

Comment:
Query only reads so no need for a transaction.

```suggestion
	return ss.WithDbSession(ctx, func(sess *DBSession) error {
```